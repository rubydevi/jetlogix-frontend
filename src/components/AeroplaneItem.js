import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import useAuth from '../hooks/useAuth';
import { showAeroplane } from '../redux/aeroplanes/aeroplanesActions';

const AeroplaneItem = ({ aeroplane, classNames }) => {
  const { auth } = useAuth();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleDispatch = (aeroplaneId) => {
    dispatch(showAeroplane({ userId: auth.id, aeroplaneId }));
    navigate('/details');
  };

  return (
    <li style={{ width: '18rem' }}>
      <button
        onClick={() => handleDispatch(aeroplane.id)}
        className={classNames.button}
        type="button"
      >
        <div className={classNames.aeroplaneBody}>
          <img
            src={aeroplane.image}
            alt={aeroplane.name}
            className={classNames.image}
          />
          <h6 className={classNames.title}>{aeroplane.name}</h6>
          <p className={classNames.description}>{aeroplane.description}</p>
        </div>
      </button>
    </li>
  );
};

AeroplaneItem.propTypes = {
  aeroplane: PropTypes.shape({
    id: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
  classNames: PropTypes.shape({
    button: PropTypes.string,
    aeroplaneBody: PropTypes.string,
    image: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
  }).isRequired,
};

export default AeroplaneItem;
