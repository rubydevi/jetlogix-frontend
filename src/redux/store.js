import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservation: reservationsReducer,
  },
});

export default store;
