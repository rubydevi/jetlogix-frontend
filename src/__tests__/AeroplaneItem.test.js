import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import AeroplaneItem from '../components/AeroplaneItem';

test('renders AeroplaneItem component correctly', () => {
  const aeroplaneData = {
    id: 1,
    image: 'some-image-url',
    name: 'Aeroplane 1',
    description: 'Description of Aeroplane 1',
  };

  const classNamesData = {
    button: 'some-button-class',
    aeroplaneBody: 'some-aeroplane-body-class',
    imageContainer: 'some-image-container-class',
    image: 'some-image-class',
    title: 'some-title-class',
    description: 'some-description-class',
  };

  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <AeroplaneItem aeroplane={aeroplaneData} classNames={classNamesData} />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
