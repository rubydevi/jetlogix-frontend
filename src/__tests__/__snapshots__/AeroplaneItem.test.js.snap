// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AeroplaneItem component correctly 1`] = `
<div>
  <li
    style="width: 18rem;"
  >
    <button
      class="some-button-class"
      type="button"
    >
      <div
        class="some-aeroplane-body-class"
      >
        <div
          class="some-image-container-class"
        >
          <img
            alt="Aeroplane 1"
            class="some-image-class"
            src="some-image-url"
          />
        </div>
        <h6
          class="some-title-class"
        >
          Aeroplane 1
        </h6>
        <p
          class="some-description-class"
        >
          Description of Aeroplane 1
        </p>
      </div>
    </button>
  </li>
</div>
`;
